### PR TITLE
Pca

### PR DESCRIPTION
**Pull request by lennart-g**
Added function to reconstruct 28×28 pixel images from the dimension reduced images.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -2,8 +2,8 @@
 import src.image_operations as image_operations
 import src.pca as pca
 import src.load_image_vectors as load_image_vectors
-import src.pickle_operations as pickle_io
 
+import src.pickle_operations as pickle_io
 tests_count = 0
 tests_success = 0
```
Comment (SakulV) [inline src/main.py]: not needed anymore

File: src/main.py [hunk 2]
```diff
@@ -45,12 +45,16 @@ def get_success_rate():
...
     print("Successfully loaded images from pickle files")
-    
+
     # Get reduces training and test images as tuple - reduced_images[0] is train_list, [1] is test_list without digits
-    reduced_images = pca.reduce_dimensions([csv_image.image for csv_image in training_lists], [csv_image.image for csv_image in test_lists])
+    reduced_images = pca.reduce_dimensions([csv_image.image for csv_image in training_lists], [csv_image.image for csv_image in test_lists], 784)
     print("PCA finished successfully")
-
+    for i in [10, 20, 50, 100, 200, 400, 784]:
+        pca.increase_dimensions(reduced_images[2], [red[:i] for red in reduced_images[1]], i)
```
Comment (SakulV) [inline src/main.py]: draw the pictures next to each other and display the according value for n

File: src/main.py [hunk 3]
```diff
@@ -60,6 +64,17 @@ def get_success_rate():
...
+    # # Save created CsvImage lists in pickle files
+    # save_pickles(training_lists, "../data/red_training.dat")
+    # save_pickles(test_lists, "../data/red_test.dat")
+    #
+    # # # Open CsvImage lists from pickle files - lowers loading time by factor 10
+    # training_lists = [None] * 2
+    # training_lists[0], training_lists[1] = load_pickles("../data/red_training.dat")
+    # test_lists = [None] * 2
+    # test_lists[0], test_lists[1] = load_pickles("../data/red_test.dat")
+    # print("Successfully loaded images from pickle files")
```
Comment (SakulV) [inline src/main.py:54]: we dont have tupels anymore

**Review by SakulV — APPROVED** [review]

File: src/main.py (hunk 1 above)
Comment (lennart-g) [reply src/main.py]: why? the main still reads the images from pickle files

File: src/main.py (hunk 2 above)
Comment (lennart-g) [reply src/main.py]: done

File: src/main.py (hunk 3 above)
Comment (lennart-g) [reply src/main.py:54]: fixed

**Review by yardenko — APPROVED** [review]

**Review by ChrisHeideb — APPROVED** [review]

**Pull request merged**